### ISSUE TITLE
Re-organize System/Device class interface for plugins and applications

### ISSUE THREAD

**Issue by shakthi-prashanth-m** [dronecore/DroneCore#325]
Header `system.h` gets installed along with `dronecore.h` and plugin header files.
DroneCore apps do:
```cpp
System &system = dc.system();
...
// have access to entire System interface
```
This implies, we're letting applications access to `System` interface as well. In other words, _applications can even compose MAVLink messages and send it across!_
I think thats not right. We may decide to export minimal interface like, `has_camera(1)`, `has_gimbal()`, etc;

As I understand, we insist DroneCore apps to use plugins rather than `System` interface. Please correct me.

On the other hand, `System` interface is needed for plugins. So, one possible way is to expose:
```cpp
class MAVLinkSystem {
// Current Device contents which are required by plugins goes here.
};
```
to plugins.
**Note:** _I chose name `MAVLinkSystem` here because its internal to DroneCore and not exposed to public. I am fine with any suitable name that makes sense._

For DroneCore applications, lets provide 
```cpp
class System {
// Minimal interface needed for DroneCore apps
};
```
Kindly share your thoughts.

**Comment by hamishwillee** [comment]
I tend to agree with you. I don't mind that app authors can use it, I mind that if they do so an this is a "public interface" we're gong to effectively be making compatibility promises about that interface going forward.

**Comment by shakthi-prashanth-m** [comment]
@julianoes @JonasVautherin @hamishwillee 
I've got these changes (on top of #310) right now (Tested). I shall submit.

**Comment by shakthi-prashanth-m** [comment]
Current design: (Details omitted)
![dronecore_current_design](https://user-images.githubusercontent.com/26615772/37587313-fb1fb6e4-2b84-11e8-9b8d-09b2bcc6cb55.png)

Proposed design: (Details omitted)
![dronecore_proposed_changes](https://user-images.githubusercontent.com/26615772/37587348-173f7832-2b85-11e8-82fd-41f01bd60722.png)
